### PR TITLE
Rewrite async.iteration to use await/async

### PR DESCRIPTION
**Pull request by justinfagnani**

### CODE REVIEW

File: lib/src/async/iteration.dart [hunk 1]
```diff
@@ -49,15 +49,13 @@ Future _doWhileAsync(
...
  */
-Future reduceAsync(Iterable iterable, initialValue, AsyncCombiner combine) =>
-    _reduceAsync(iterable.iterator, initialValue, combine);
-
-Future _reduceAsync(Iterator iterator, currentValue, AsyncCombiner combine) {
-  if (iterator.moveNext()) {
-    return combine(currentValue, iterator.current)
-        .then((result) => _reduceAsync(iterator, result, combine));
+Future reduceAsync(Iterable iterable, initialValue, AsyncCombiner combine) async
+{
```
Comment (yjbanov) [inline lib/src/async/iteration.dart]: Move `{` to the previous line

Comment (justinfagnani) [inline lib/src/async/iteration.dart]: it's > 80 cols... where's the formatter say I should put it?

Comment (yjbanov) [inline lib/src/async/iteration.dart]: Hmm, try it and see what it says.

**Comment by yjbanov** [comment]
LGTM

**Comment by justinfagnani** [comment]
I'm not so sure we should merge right away, I was just seeing where the code could be more clear in 1.9. We need to bump the quiver version and SDK constraint. I'm not sure if there are other things we can do in 1.9 (generators?), so maybe this deserves a 1.9 branch...

**Comment by zoechi** [comment]
Should also increase the lower bound for `sdk` to at least `1.9.0`
```yaml
environment:
  sdk: '>=1.9.0 <2.0.0'
```

**Pull request closed without merging**